statsmodels/genmod/tests/test_glm.py

```diff
@@ -1318,16 +1365,245 @@ def test_wtd_patsy_missing():
+    def __init__(self):
+        from statsmodels.datasets.cpunish import load_pandas
+        self.data = load_pandas()
+        self.exog = self.data.exog[['INCOME', 'SOUTH']]
+        self.endog = self.data.endog
+
+    def test_resid(self):
+        l = len(self.res1.resid_response) - 1
+        l2 = len(self.res2.resid_response) - 1
+        assert_allclose(np.concatenate((self.res1.resid_response[:17],
+                                        [self.res1.resid_response[l]])),
+                        np.concatenate((self.res2.resid_response[:17],
+                                        [self.res2.resid_response[l2]])),
+                        rtol=1e-5, atol=1e-5)
+        assert_allclose(np.concatenate((self.res1.resid_pearson[:17],
+                                        [self.res1.resid_pearson[l]])),
+                        np.concatenate((self.res2.resid_pearson[:17],
+                                        [self.res2.resid_pearson[l2]])),
+                        rtol=1e-5, atol=1e-5)
+        assert_allclose(np.concatenate((self.res1.resid_deviance[:17],
+                                        [self.res1.resid_deviance[l]])),
+                        np.concatenate((self.res2.resid_deviance[:17],
+                                        [self.res2.resid_deviance[l2]])),
+                        rtol=1e-5, atol=1e-5)
+        # Not working either...
+        if not isinstance(self, (TestTweedieLog1, TestTweedieLog15Fair)):
+            assert_allclose(np.concatenate((self.res1.resid_working[:17],
+                                            [self.res1.resid_working[l]])),
+                            np.concatenate((self.res2.resid_working[:17],
+                                            [self.res2.resid_working[l2]])),
+                            rtol=1e-5, atol=1e-5)
+
+    def test_bse(self):
+        assert_allclose(self.res1.bse, self.res2.bse, atol=1e-6, rtol=1e6)
+
+    def test_params(self):
+        assert_allclose(self.res1.params, self.res2.params, atol=1e-5,
+                        rtol=1e-5)
+
+    def test_deviance(self):
+        assert_allclose(self.res1.deviance, self.res2.deviance, atol=1e-6,
+                        rtol=1e-6)
+
+    def test_df(self):
+        assert_equal(self.res1.df_model, self.res2.df_model)
+        assert_equal(self.res1.df_resid, self.res2.df_resid)
+
+    def test_fittedvalues(self):
+        l = len(self.res1.fittedvalues) - 1
+        l2 = len(self.res2.resid_response) - 1
+        assert_allclose(np.concatenate((self.res1.fittedvalues[:17],
+                                        [self.res1.fittedvalues[l]])),
+                        np.concatenate((self.res2.fittedvalues[:17],
+                                        [self.res2.fittedvalues[l2]])),
+                        atol=1e-4, rtol=1e-4)
+
+    def test_summary(self):
+        self.res1.summary()
+        self.res1.summary2()
+
+
+class TestTweediePower15(CheckTweedie):
+    def __init__(self):
```

Comment:
convert all the `__init__` to a classmethod `setup_class`   
check other tests for the format

this "should" speed up the unit tests because nose runs `__init__` separately for each method of the class, i.e. this currently estimates the same thing seven times.